src/synth/fluid_synth.c

```diff
@@ -2383,6 +2383,16 @@ fluid_synth_set_sample_rate(fluid_synth_t* synth, float sample_rate)
   fluid_synth_update_mixer(synth, fluid_rvoice_mixer_set_samplerate, 
 			   0, sample_rate);
   fluid_synth_api_exit(synth);
```

Comment:
I'm afraid we are giving up our lock too early here. Consider this thread gets suspended right after  `fluid_synth_api_exit()` and the next call would be `delete_fluid_synth()` deleting `synth->ladspa_fx`.

On the other hand I'm not sure if we might experience a deadlock waiting to enter LADSPA_API.

Reply:
And ofc if you're postponing `fluid_synth_api_exit()` so `ladspa_set_sample_rate` is protected you can ignore my first comment.

Reply:
Yes, it deadlocks if the fluid API is not released before calling fluid_ladspa_set_sample_rate. Hm...

Reply:
Ah... we could handle this via the mixer event queue as well. Then there would be no deadlock. I'll draft the idea in a commit.